packages/client/foreman_ygg_worker/foreman_ygg_worker.spec

```diff
@@ -43,7 +43,13 @@ BuildRequires: go
 %else
 BuildRequires: golang
 %endif
+
+# Use rich dependencies if available
+%if 0%{?rhel} >= 8 || 0%{?fedora}
```

Comment:
I think we could avoid the if-else and just lock it down? Since there is an inherent technology difference between the versions.

Reply:
On EL7 rpm doesn't support rich dependencies and I don't think you can properly narrow it. The best you can do is add 2 `Requires` statements which rpm will evaluate separately.

@evgeni any thoughts?

Reply:
Yeah, this. Or:
```
Requires: yggdrasil >= 0.2
Conflicts: yggdrasil >= 0.3
```

Which is technically more correct.

But I am fine with using rich deps and ignoring EL7 for those last breaths it still has (it won't get yggdrasil 0.3+ anyway, RIGHT?!).

Reply:
I now added that in the `else` branch. I still think the rich dependency helps rpm, but that's also just speculation from my side.